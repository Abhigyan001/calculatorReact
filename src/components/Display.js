import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ result }) => (
  <div id="display">    
    <p className="result">{result}</p>    
  </div>
);

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};

export default Display;
