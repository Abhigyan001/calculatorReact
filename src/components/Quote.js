/* eslint-disable max-len */
import React from 'react';
import styles from './styles.module.css';

function Quote() {
  return <div className={styles.quote}>Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“ William Paul Thurston</div>;
}

export default Quote;
