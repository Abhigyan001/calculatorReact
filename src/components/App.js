/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import buttons from '../buttons/buttons';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  handleClick = buttonName => {
    const newState = calculate(this.state, buttonName);
    this.setState(newState);
  };

  keyDownListener = e => {
    let buttonName = e.key;
    if (buttonName === 'Enter') buttonName = '=';
    if (buttonName === '/') buttonName = 'รท';
    if (buttonName === '*') buttonName = 'X';
    const validButton = buttons.some(row => row.includes(buttonName));
    if (validButton) {
      this.handleClick(buttonName);
    }
  };

  componentDidMount() {
    document.addEventListener('keydown', this.keyDownListener);
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.keyDownListener);
  }

  handleClick = buttonName => {
    const newState = calculate(this.state, buttonName);
    this.setState(newState);
  };

  render() {
    const { total, next, operation } = this.state;
    return (
      <div>
        <Display total={total} next={next} operation={operation} />
        <ButtonPanel data={this.state} buttons={buttons} clickHandler={this.handleClick} />
      </div>
    );
  }
}

export default App;
