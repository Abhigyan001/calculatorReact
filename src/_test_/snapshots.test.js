import renderer from 'react-test-renderer';
import App from '../components/App';
import Display from '../components/Display';
import Home from '../components/Home';
import Quote from '../components/Quote';

it('renders App correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Display correctly', () => {
  const tree = renderer
    .create(<Display />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Home correctly', () => {
  const tree = renderer
    .create(<Home />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders Quote correctly', () => {
  const tree = renderer
    .create(<Quote />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
