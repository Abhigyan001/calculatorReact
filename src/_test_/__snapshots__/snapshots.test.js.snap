// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div
  className="calculator"
>
  <h1
    className="math"
  >
    Let’s do some math!
  </h1>
  <div
    className="app"
  >
    <div
      className="display"
    >
      <p>
        0
      </p>
      <span
        className="displaySpan"
      >
         
      </span>
    </div>
    <div
      className="button_panel"
    >
      <div
        className="button_panel_row"
      >
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button
                  null
                  null"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
      </div>
      <div
        className="button_panel_row"
      >
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button
                  null
                  null"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
      <div
        className="button_panel_row"
      >
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button
                  null
                  null"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="button_panel_row"
      >
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button
                  null
                  null"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="button_panel_row"
      >
        <button
          className="button
                  button_gray
                  button_double"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button
                  button_gray
                  null"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button
                  null
                  null"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders Display correctly 1`] = `
<div
  className="display"
>
  <p>
    0
  </p>
  <span
    className="displaySpan"
  >
     
  </span>
</div>
`;

exports[`renders Home correctly 1`] = `
<div
  className="home"
>
  <h1>
    Welcome to our page!
  </h1>
  <p
    className="para"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum.
  </p>
  <p
    className="para"
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Utenim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui o!cia deserunt mollit anim id est laborum.
  </p>
</div>
`;

exports[`renders Quote correctly 1`] = `
<div
  className="quote"
>
  Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. – William Paul Thurston
</div>
`;
